### PR TITLE
HTCONDOR-1487-singularity-regression-test

### PR DESCRIPTION
**Pull request by GregThain**
[HTCONDOR-1487](https://opensciencegrid.atlassian.net/browse/HTCONDOR-1487)

Add regression test that runs a container universe job with a sif file under singularity, if singularity works  on the platform.

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

[HTCONDOR-1487]: https://opensciencegrid.atlassian.net/browse/HTCONDOR-1487?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by timtheisen** [comment]
@GregThain apptainer is now present in all CentOS, AlmaLinux, and Fedora BaTLab build images.

File: src/condor_tests/test_singularity_sif.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+# jobs.  If not, skip the test.
+
+# Then, stand up a condor with SINGULARITY_BIND_EXPR set to all
+# the /bin, /usr/bin and /lib directories from the host, so that
+# we can run any binary from the host inside the container without
+# worrying about also copying in shared libraries, helper files,
+# config files etc.  
+
+# The first test is to just cat /singularity, which is a file which 
+# will only exist inside the container. If that exits non-zero,
```
Comment (Todd-L-Miller) [inline src/condor_tests/test_singularity_sif.py:20]:
```suggestion
# will only exist inside the container. If that exits zero,
```
... I hope.

File: src/condor_tests/test_singularity_sif.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        timeout=60,
+        verbose=True,
+        fail_condition=ClusterState.any_held,
+    )
+    return ctj.query()[0]
+
+# For the test to work, we need a singularity/apptainer which can work with
+# SIF files, which is any version of apptainer, or singularity >= 3
+def SingularityIsWorthy():
+    result = subprocess.run("ssssingularity --version", shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```
Comment (Todd-L-Miller) [inline src/condor_tests/test_singularity_sif.py:91]:
```suggestion
    result = subprocess.run("singularity --version", shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

**Review by Todd-L-Miller — COMMENTED** [review]
Discussed in-person, will re-review after changes.

**Comment by Todd-L-Miller** [comment]
(The idea for the current patch is that it's hard to inspect the exit code of the job directly, so instead have the job go idle and the test time out instead of just failing.  This is obviously sub-optimal, and I'll look into why the exit code of the job is hard to get.)

**Review by Todd-L-Miller — APPROVED** [review]
LGTM.  Please make sure it passes the BaTLab tests, etc.

**Pull request merged**